Struct field that is a function that returns a function panics

### The following program `sample.go` triggers an unexpected result

```go
package main

import "fmt"

type foobar struct {
	callback func(string) func()
}

func cb(text string) func() {
	return func() {
		fmt.Println(text)
	}
}

func main() {
	// These ways of invoking it all work...
	cb("Hi from inline callback!")()

	asVarTest1 := cb("Hi from asVarTest1 callback!")
	asVarTest1()

	asVarTest2 := cb
	asVarTest2("Hi from asVarTest2 callback!")()

	// But inside a struct paincs in yaegi...
	asStructField := &foobar{callback: cb}
	asStructField.callback("Hi from struct field callback!")() // <--- panics here
}

// Output:
// Hi from inline callback!
// Hi from asVarTest1 callback!
// Hi from asVarTest2 callback!
// Hi from struct field callback!
```


### Expected result

```console
$ go run sample.go
Hi from inlined callback!
Hi from asVar1 callback!
Hi from asVar2 callback!
Hi from foobar callback!
```


### Got

```console
$ yaegi sample.go
Hi from inline callback!
Hi from asVarTest1 callback!
Hi from asVarTest2 callback!
calling_func_func_field_in_struct_panics.go:27:2: panic
run: reflect.Set: value of type func() is not assignable to type *interp.node
goroutine 1 [running]:
runtime/debug.Stack(0x1, 0xc000304800, 0x40)
        /usr/local/Cellar/go/1.16.5/libexec/src/runtime/debug/stack.go:24 +0x9f
github.com/traefik/yaegi/interp.(*Interpreter).eval.func1(0xc0001c5c18)
        /Users/johnny/go/pkg/mod/github.com/traefik/yaegi@v0.9.22-0.20210721094404-2f8493c40519/interp/interp.go:504 +0xc8
panic(0x1a17cc0, 0xc0001af700)
        /usr/local/Cellar/go/1.16.5/libexec/src/runtime/panic.go:965 +0x1b9
github.com/traefik/yaegi/interp.runCfg.func1(0xc00015edc0, 0xc0004a1a00, 0xc0001c59f8)
        /Users/johnny/go/pkg/mod/github.com/traefik/yaegi@v0.9.22-0.20210721094404-2f8493c40519/interp/run.go:185 +0x253
panic(0x1a17cc0, 0xc0001af700)
        /usr/local/Cellar/go/1.16.5/libexec/src/runtime/panic.go:965 +0x1b9
reflect.Value.assignTo(0x1a0f260, 0xc00012af10, 0x93, 0x1b5824a, 0xb, 0x1af22c0, 0x0, 0xc00012af30, 0x1, 0x1)
        /usr/local/Cellar/go/1.16.5/libexec/src/reflect/value.go:2451 +0x3f7
reflect.Value.Set(0x1af22c0, 0xc000134a10, 0x196, 0x1a0f260, 0xc00012af10, 0x93)
        /usr/local/Cellar/go/1.16.5/libexec/src/reflect/value.go:1564 +0xbd
github.com/traefik/yaegi/interp.callBin.func10(0xc00015edc0, 0xc00010fcc0)
        /Users/johnny/go/pkg/mod/github.com/traefik/yaegi@v0.9.22-0.20210721094404-2f8493c40519/interp/run.go:1500 +0x25c
github.com/traefik/yaegi/interp.runCfg(0xc0004a1a00, 0xc00015edc0)
        /Users/johnny/go/pkg/mod/github.com/traefik/yaegi@v0.9.22-0.20210721094404-2f8493c40519/interp/run.go:191 +0x87
github.com/traefik/yaegi/interp.(*Interpreter).run(0xc000386000, 0xc0004a1200, 0xc00015ec60)
        /Users/johnny/go/pkg/mod/github.com/traefik/yaegi@v0.9.22-0.20210721094404-2f8493c40519/interp/run.go:122 +0x2b0
github.com/traefik/yaegi/interp.(*Interpreter).eval(0xc000386000, 0xc00030b080, 0x2a7, 0x7ffeefbffbce, 0x2b, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /Users/johnny/go/pkg/mod/github.com/traefik/yaegi@v0.9.22-0.20210721094404-2f8493c40519/interp/interp.go:596 +0x80c
github.com/traefik/yaegi/interp.(*Interpreter).EvalPath(0xc000386000, 0x7ffeefbffbce, 0x2b, 0xc00030ab00, 0x2a7, 0x0, 0x0, 0xc0001c5cb0)
        /Users/johnny/go/pkg/mod/github.com/traefik/yaegi@v0.9.22-0.20210721094404-2f8493c40519/interp/interp.go:418 +0x105
main.runFile(0xc000386000, 0x7ffeefbffbce, 0x2b, 0x0, 0x1, 0x0)
        /Users/johnny/go/pkg/mod/github.com/traefik/yaegi@v0.9.22-0.20210721094404-2f8493c40519/cmd/yaegi/run.go:149 +0xb1
main.run(0xc000138010, 0x1, 0x1, 0xc00010c101, 0xc000102480)
        /Users/johnny/go/pkg/mod/github.com/traefik/yaegi@v0.9.22-0.20210721094404-2f8493c40519/cmd/yaegi/run.go:112 +0x9c5
main.main()
        /Users/johnny/go/pkg/mod/github.com/traefik/yaegi@v0.9.22-0.20210721094404-2f8493c40519/cmd/yaegi/yaegi.go:144 +0x415
```


### Yaegi Version

2f8493c40519

### Additional Notes

_No response_

--- a/_test/issue-1202.go
+++ b/_test/issue-1202.go
@@ -1,0 +1,34 @@
+package main
+
+import "fmt"
+
+type foobar struct {
+	callback func(string) func()
+}
+
+func cb(text string) func() {
+	return func() {
+		fmt.Println(text)
+	}
+}
+
+func main() {
+	// These ways of invoking it all work...
+	cb("Hi from inline callback!")()
+
+	asVarTest1 := cb("Hi from asVarTest1 callback!")
+	asVarTest1()
+
+	asVarTest2 := cb
+	asVarTest2("Hi from asVarTest2 callback!")()
+
+	// But inside a struct panics in yaegi...
+	asStructField := &foobar{callback: cb}
+	asStructField.callback("Hi from struct field callback!")() // <--- panics here
+}
+
+// Output:
+// Hi from inline callback!
+// Hi from asVarTest1 callback!
+// Hi from asVarTest2 callback!
+// Hi from struct field callback!

--- a/interp/run.go
+++ b/interp/run.go
@@ -1497,7 +1497,11 @@ func callBin(n *node) {
 				}
 				out := callFn(value(f), in)
 				for i := 0; i < len(out); i++ {
-					getFrame(f, n.level).data[n.findex+i].Set(out[i])
+					if out[i].Type().Kind() == reflect.Func {
+						getFrame(f, n.level).data[n.findex+i] = out[i]
+					} else {
+						getFrame(f, n.level).data[n.findex+i].Set(out[i])
+					}
 				}
 				return tnext
 			}

